backend/eventd/entity.go

```diff
@@ -0,0 +1,55 @@
+package eventd
+
+import (
+	"context"
+
+	corev2 "github.com/sensu/sensu-go/api/core/v2"
+	"github.com/sensu/sensu-go/backend/store"
+)
+
+// createProxyEntity creates a proxy entity for the given event if the entity
+// does not exist already and returns the entity created
+func createProxyEntity(event *corev2.Event, s store.EntityStore) error {
+	entityName := event.Entity.Name
```

Comment:
I think this needs a nil check. If event.Entity is nil, then this will cause a panic.

The feature is kind of weird in general because the idea is that you don't need an entity... but you actually do, to specify its name.

Do we need to look at automatic proxy entity naming as a feature?

Reply:
An entity and its name are actually required, since we validate the event just before calling this function: 

https://github.com/sensu/sensu-go/blob/59af146941d78108d4a96864122e498693ef5caa/backend/eventd/eventd.go#L236-L238

There are two use cases for this function:

1. A `proxy_entity_name` is configured in the check, therefore we need to replace the event's entity, which refers to the agent's entity that executed the check, with a brand new entity named using this `proxy_entity_name` attribute as its name
2. The user submitted an event via the backend API, in which case we take whatever entity was provided to create the proxy entity.

Makes sense?